cmd/promtool/main.go

```diff
@@ -114,8 +119,12 @@ func main() {
 	checkConfigSyntaxOnly := checkConfigCmd.Flag("syntax-only", "Only check the config file syntax, ignoring file and content validation referenced in the config").Bool()
 	checkConfigLint := checkConfigCmd.Flag(
 		"lint",
-		"Linting checks to apply to the rules specified in the config. Available options are: "+strings.Join(lintOptions, ", ")+". Use --lint=none to disable linting",
+		"Linting checks to apply to the rules/scrape configs specified in the config. Available options are: "+strings.Join(lintConfigOptions, ", ")+". Use --lint=none to disable linting",
 	).Default(lintOptionDuplicateRules).String()
+	checkConfigLintLookbackDelta := checkConfigCmd.Flag(
+		"lint.lookback-delta",
+		"The maximum lookback duration. This config is only used for scrape config linting.",
```

Comment:
I'm in two minds about having this as a `--lint.` specific option, mostly because I don't know if users will have enough context to understand the implication of "This config is only used for scrape config linting".

Currently it isn't possible to unit test something with a different lookback delta. Why not just add `--query.lookback-delta` to promtool (maybe at the `check` level?)

This would be consistent if we wanted more options mirroring the relevant `--query` options. For example `--query.max-samples` could be relevant in unit testing too (although probably not, as it would be a lot of data to load into a test).

Reply:
I pushed a revision moving the flag to `promtool check --query.lookback-delta`. PTAL.

Reply:
Yes, looks good.